lib/beaker-pe/pe-client-tools/install_helper.rb

```diff
@@ -0,0 +1,74 @@
+module Beaker
+  module DSL
+    module InstallUtils
+      module PEClientTools
+
+        def install_pe_client_tools_on(hosts, opts = {})
+          # FIXME: accomodate production released location(s)
+          #{{{
+          product = 'pe-client-tools'
+          required_keys = [:puppet_collection, :pe_client_tools_sha, :pe_client_tools_version]
+
+          unless required_keys.all? { |opt| opts.keys.include?(opt) }
+            raise ArgumentError, "The keys #{required_keys.to_s} are required in the opts hash"
+          end
+          urls = { :dev_builds_url   => "http://builds.delivery.puppetlabs.net",
+          }
+
+          opts = urls.merge(opts)
+
+          block_on hosts do |host|
+            package_name = nil
+            variant, version, arch, codename = host['platform'].to_array
+            case host['platform']
+            when /el-|fedora|sles|centos|cisco_/
+              release_path = "#{opts[:dev_builds_url]}/#{product}/#{ opts[:pe_client_tools_sha] }/artifacts/#{variant}/#{version}/#{opts[:puppet_collection]}/#{arch}"
+              package_name = product.dup
+              package_name << "-#{opts[:pe_client_tools_version]}-1.#{variant}#{version}.#{arch}.rpm" if opts[:pe_client_tools_version]
+            when /debian|ubuntu|cumulus|huaweios/
+              release_path = "#{opts[:dev_builds_url]}/#{product}/#{ opts[:pe_client_tools_sha] }/artifacts/deb/#{codename}/#{opts[:puppet_collection]}"
+              package_name = product.dup
+              package_name << "_#{opts[:pe_client_tools_version]}-1#{host['platform'].codename}_#{arch}.deb" if opts[:pe_client_tools_version]
+            when /windows/
+              release_path = "#{opts[:dev_builds_url]}/#{product}/#{ opts[:pe_client_tools_sha] }/artifacts/#{variant}/#{opts[:puppet_collection]}/x#{arch}"
+              package_name = product.dup
+              package_name << "-#{opts[:pe_client_tools_version]}.1-x#{arch}_VANAGON.msi" if opts[:pe_client_tools_version]
+            when /osx/
+              release_path = "#{opts[:dev_builds_url]}/#{product}/#{ opts[:pe_client_tools_sha] }/artifacts/apple/#{version}/#{opts[:puppet_collection]}/#{arch}"
+              package_base = product.dup
+              package_base << "-#{opts[:pe_client_tools_version]}" if opts[:pe_client_tools_version]
+              package_name = package_base.dup
+              package_name << '-1' if opts[:pe_client_tools_version]
+              installer    = package_name + '-installer.pkg'
+              package_name << ".#{variant}#{version}.dmg" if opts[:pe_client_tools_version]
+            else
+              raise "install_puppet_agent_on() called for unsupported " +
+                "platform '#{host['platform']}' on '#{host.name}'"
+            end
```

Comment:
I would suggest refactoring this section that's really looking to get host-specific install information into a helper method.  `install_puppet_agent_pe_promoted_repo_on` has been refactored in this way in [lines 1171-81](https://github.com/puppetlabs/beaker/blob/master/lib/beaker/dsl/install_utils/foss_utils.rb#L1177-L1181), and I think it's made the method much more understandable and maintainable.

It's not as good that you won't be able to make it a host method at this time, but a helper in this same file containing these host-specific details would still improve this method quite a bit, and has the advantage that having it here rather than a host method is closer to you & your code, and as such future changes will be easier to make than in beaker-core.

To combat the issue that there's an additional `installer` variable needed for osx, that variable is only used once, and it's in a line that's already mac-specific, so I would suggest just refactoring the osx install line to use
```ruby
"#{package_name}-installer.pkg"
```
as the parameter instead of needing the `installer` variable.